### PR TITLE
Fix theme support for ablog

### LINKED ISSUE #294: Theme support for partial overwriting Ablog templates is maybe broken
### Describe the bug

I am trying to use recent template rework for overwriting `ablog/postcard.html` only but keep the rest of Ablog templates. However, I can't find out the way to manage it. **Is there some known theme which ships their Ablog templates?** Maybe it would help me.

### To Reproduce

However, I suspect that this template rework is broken.

1. Because I want to "hide" only some Ablog templates, I don't set `skip_injecting_base_ablog_templates = True`.
2. Under my theme folder, I created `ablog/postcard.html` 
3. In my `theme.toml` I added
   ```
   [ablog]
   inject_templates_after_theme = true
   ```
4. My theme's `ablog/postcard.html` is never used.

There is at least one reason why not. Consider the [relevant excerpt from Ablog](https://github.com/sunpy/ablog/blob/main/src/ablog/__init__.py#L125):

```
    try:
        after_theme = theme.get_config("ablog", "inject_templates_after_theme", False)
    except ThemeError:
        after_theme = False
    if after_theme:
        loaders.append(SphinxFileSystemLoader(get_html_templates_path()))
    else:
        loaders.insert(templatepathlen, SphinxFileSystemLoader(get_html_templates_path()))
```

The `theme.get_config("ablog", "inject_templates_after_theme", False)` call raises `ThemeError` exception with message `Theme configuration sections other than [theme] and [options] are not supported`.

This exception is incorrectly interpreted as missing default value and code flow went to `else` block.

But as the message said, the Sphinx's `Theme.get_config()` doesn't allow reading custom sections like `[ablog]` from `theme.toml` files. Unfortunately in such case, it raises ThemeError too.

(Excerpt from [`Theme.get_config()`](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/theming.py#L114):)

```
           msg = __(
                'Theme configuration sections other than [theme] and [options] '
                'are not supported (tried to get a value from %r).'
            )
            raise ThemeError(msg)
```

### Screenshots

_No response_

### System Details

Sphinx==8.1.2
ablog==0.11.11

### Installation method

_No response_

### PR DESCRIPTION
**Pull request by liborjelinek**
## PR Description

- Fixes #294, theme support for partial overwriting Ablog templates.
- add tests
- add missing doc article for new options
- move sidebar theming description to this new article to have all theming docs in single place

### CODE REVIEW

File: docs/manual/templates-themes.rst [hunk 1]
```diff
@@ -0,0 +1,97 @@
+Templating and Themes Support
+=============================
+
+.. post:: Oct 26, 2024
+   :tags: themes
+   :category: Manual
+   :author: Libor
+
+Ablog, as being Sphinx extensions, it has highly customizable HTML output. The generated HTML files are based `Sphinx templates`_. You, or Sphinx themes can partially or completely override these templates to customize the resulting HTML.
```
Comment (liborjelinek) [inline docs/manual/templates-themes.rst:9]: "Extension", not plular

File: docs/manual/templates-themes.rst [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+#. Create a folder ``ablog/`` inside the ``_templates/`` folder.
+#. Create here a file with the same name as the template you want to override. For example, ``postcard.html``. This file will be used as a custom template for the sidebar. You can copy the content of the original template from the Ablog source code and modify it as you need.
+#. Optionall: if you want to complete override all Ablog templates, set the :confval:`skip_injecting_base_ablog_templates` option to ``True``, copy all Ablog templates here and customize them as you need.
+
+Customizing templates in the theme
+----------------------------------
+
+If you are a Sphinx theme author, you can ship customized Ablog templates in your theme. You can override Ablog templates by placing them in the ``ablog/`` folder in your theme templates, e.g., ``ablog/postcard.html``.
+
+#. In theme root (where is the ``theme.conf`` file), create a folder ``ablog/``.
```
Comment (liborjelinek) [inline docs/manual/templates-themes.rst:88]: theme.toml

File: docs/manual/templates-themes.rst [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+#. Create here a file with the same name as the template you want to override. For example, ``postcard.html``.
+#. This file will be used as a custom template for the sidebar. You can copy the content of the original template from the Ablog source code and modify it as you need.
+#. In your ``theme.conf`` file, add the following (under the ``[options]`` section):
```
Comment (liborjelinek) [inline docs/manual/templates-themes.rst:91]: theme.toml

File: docs/manual/templates-themes.rst [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+   .. code-block:: ini
```
Comment (liborjelinek) [inline docs/manual/templates-themes.rst:93]: toml mode, not ini

File: docs/release/ablog-v0.11-released.rst [hunk 5]
```diff
@@ -29,6 +29,13 @@ Pull Requests merged in:
 
 `Add external links for posts <https://github.com/sunpy/ablog/pull/112>`__ from `Chris Holdgraf <https://github.com/choldgraf>`__.
 
+Unreleased
+----------
+
+Pull Requests merged in:
+
+`Fix theme support for Ablog <https://github.com/sunpy/ablog/pull/XXX>`__ from `Libor Jelínek <https://github.com/liborjelinek/>`__.
```
Comment (liborjelinek) [inline docs/release/ablog-v0.11-released.rst:37]: Change to this PR URL

File: src/ablog/tests/test_templates.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
+from pathlib import Path
+import pytest
+from sphinx.application import Sphinx
+from sphinx.errors import ThemeError
+
+
+def read_text(path):
```
Comment (liborjelinek) [inline src/ablog/tests/test_templates.py:7]: Uncessesary

**Review by nabobalis — APPROVED** [review]

**Comment by nabobalis** [comment]
Thanks again @liborjelinek

**Pull request merged**